javav2/example_code/pinpoint/src/main/java/com/example/pinpoint/ExportEndpoints.java

```diff
@@ -0,0 +1,228 @@
+//snippet-sourcedescription:[ExportEndpoints.java demonstrates how to export endpoints to an Amazon S3 bucket.]
+//snippet-keyword:[AWS SDK for Java v2]
+//snippet-keyword:[Code Sample]
+//snippet-keyword:[Amazon Pinpoint]
+//snippet-sourcetype:[full-example]
+//snippet-sourcedate:[12/05/2020]
+//snippet-sourceauthor:[scmacdon-aws]
+
+/*
+   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0
+*/
+
+package com.example.pinpoint;
+
+//snippet-start:[pinpoint.java2.export_endpoint.import]
+import software.amazon.awssdk.core.ResponseBytes;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.pinpoint.PinpointClient;
+import software.amazon.awssdk.services.pinpoint.model.ExportJobRequest;
+import software.amazon.awssdk.services.pinpoint.model.PinpointException;
+import software.amazon.awssdk.services.pinpoint.model.CreateExportJobRequest;
+import software.amazon.awssdk.services.pinpoint.model.CreateExportJobResponse;
+import software.amazon.awssdk.services.pinpoint.model.GetExportJobResponse;
+import software.amazon.awssdk.services.pinpoint.model.GetExportJobRequest;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.GetObjectRequest;
+import software.amazon.awssdk.services.s3.model.ListObjectsV2Request;
+import software.amazon.awssdk.services.s3.model.ListObjectsV2Response;
+import software.amazon.awssdk.services.s3.model.S3Object;
+import software.amazon.awssdk.services.s3.model.GetObjectResponse;
+import software.amazon.awssdk.services.s3.model.S3Exception;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+//snippet-end:[pinpoint.java2.export_endpoint.import]
+
+/**
+ *  To run this code example, you need to create an IAM role with the correct policy as described in this documentation:

```

Comment:
Spell out "AWS Identity and Access Management (IAM)" on "first mention" in a code file.

Reply:
fixed